### PR TITLE
fix: should trim in data url.

### LINKED ISSUE #2300: [Bug]: errors when spaces exists after `data:*;base64,`
### Reproduction link or steps

1. Open https://stackblitz.com/edit/github-6jajke-ybzkfq?file=src%2Fentry.js
1. `npm run build` runs automatically.
1. An error happens

### What is expected?

No error happens

### What is actually happening?

The following error happens:

> Rolldown internal error: Base64Error

### System Info

```Shell
Stackblitz

  System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.20.3 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    rolldown: https://pkg.pr.new/rolldown@351d2c7 => 0.13.2
```

### Any additional comments?

Faced this while trying rolldown with Vite.

### PR DESCRIPTION
**Pull request by 7086cmd**
resolves #2300.

### CODE REVIEW

File: crates/rolldown/tests/rolldown/issues/2300/main.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+// space after ,
+import base64Msg from 'data:text/javascript;base64, ZXhwb3J0IGRlZmF1bHQgJ2hpJw==';
```
Comment (hyf0) [inline crates/rolldown/tests/rolldown/issues/2300/main.js:2]: Where is the space?

Comment (hyf0) [reply crates/rolldown/tests/rolldown/issues/2300/main.js:2]: Saw it

**Review by hyf0 — APPROVED** [review]
Thanks!

File: crates/rolldown/tests/rolldown/issues/2300/main.js (hunk 1 above)
Comment (IWANABETHATGUY) [reply crates/rolldown/tests/rolldown/issues/2300/main.js:2]: after `base64,`

**Pull request merged**